index.js

```diff
@@ -49,28 +50,46 @@ function validateList (list, supported, name) {
 }
 
 function getNameAndVersion (opts, dir, cb) {
-  var pkg = require(path.join(dir, 'package.json'))
+  var props = []
+  if (!opts.name) props.push(['productName', 'name'])
+  if (!opts.version) props.push(['dependencies.electron', 'devDependencies.electron'])
 
   // Name and version provided, no need to infer
-  if (opts.name && opts.version) return cb(null)
+  if (props.length === 0) return cb(null)
 
-  // Infer name from package.json
-  opts.name = opts.name || prop(pkg, 'productName') || prop(pkg, 'name')
-
-  // Infer electron version from package.json
-  var electronVersion = prop(pkg, 'dependencies.electron') || prop(pkg, 'devDependencies.electron')
-  var electronPrebuiltVersion = prop(pkg, 'dependencies.electron-prebuilt') || prop(pkg, 'devDependencies.electron-prebuilt')
-  opts.version = opts.version || electronVersion || electronPrebuiltVersion
-
-  if (!electronVersion && !electronPrebuiltVersion) return cb(null)
+  // Search package.json files to infer name and version from
+  getPackageInfo(props, dir, function (err, result) {
+    if (err) {
+      // `get-package-info` exploded looking for `electron`. Try `electron-prebuilt` instead
```

Comment:
Then it would be `props.push(['dependencies.electron', 'devDependencies.electron', 'dependencies.electron-prebuilt', 'devDependencies.electron-prebuilt'])`